test/cmdlineTests/yul_string_format_hex/output.json

```diff
@@ -12,7 +12,7 @@ object \"C_10\" {
 
         // Begin state variable initialization for contract \"C\" (0 variables)
         // End state variable initialization for contract \"C\".
-
+        if callvalue() { revert(0, 0) }
```

Comment:
Was it a bug that these were not here before?

Reply:
In oldcodegen it was calling callValueCheck liek this:
```
if (FunctionDefinition const* constructor = _contract.constructor())
	appendConstructor(*constructor);
else if (auto c = _contract.nextConstructor(m_context.mostDerivedContract()))
	appendBaseConstructor(*c);
else
	appendCallValueCheck();
```

and in Sol2Yul it was called only in case when constructor is not payable. What should be correct to do?
I assumed old codegen was doing right thing

Reply:
The check should only be there if the most derived constructor is not payable (if one of them is, then all of them should).

Reply:
Ok, fixing it now then